### PR TITLE
feat(gengapic): rest-numeric-enums option enables enum-encoding sys param

### PR DESCRIPTION
**Pull request by noahdietz**
For REGAPIC API calls, when the `rest-numeric-enums` plugin option is provided, include the new system parameter variant of `$alt=json` that enables proto-json encoding of enums in server responses as integers - `enum-encoding=int`.  The default is false. Includes the bazel attribute as well.

Also updates the README with option documentation.

Requested in http://b/232457394.

### CODE REVIEW

File: internal/gengapic/genrest.go [hunk 1]
```diff
@@ -396,6 +393,8 @@ func (g *generator) generateQueryString(m *descriptor.MethodDescriptorProto) {
 	sort.Strings(fields)
 
 	p("params := url.Values{}")
+	p(`params.Add("$alt", "json;enum-encoding=int")`)
```
Comment (codyoss) [inline internal/gengapic/genrest.go:396]: Does the dollar sign have significance? We set this param for discovery clients too, but just as `alt`

Comment (noahdietz) [reply internal/gengapic/genrest.go:396]: @vchudnov-g ?

I was following what was described in the the linked bug, I'm not 100% sure tbh.

Comment (noahdietz) [reply internal/gengapic/genrest.go:396]: Ah it is documented as supporting both: https://cloud.google.com/apis/docs/system-parameters.

**Review by codyoss — APPROVED** [review]

File: internal/gengapic/genrest.go (hunk 1 above)
Comment (vchudnov-g) [reply internal/gengapic/genrest.go:396]: Google system params accept both. It seems to me it's safer to include the dollar sign, as you've done.

**Review by vchudnov-g — APPROVED** [review]

**Comment by vchudnov-g** [comment]
Oh, one question: are you planning on encoding all enums as ints when sending the request too? We think that would be desirable for symmetry.

**Comment by noahdietz** [comment]
> Oh, one question: are you planning on encoding all enums as ints when sending the request too? We think that would be desirable for symmetry.

That was already done in #994 :)

**Comment by vchudnov-g** [comment]
Oh, right! Great!

**Comment by noahdietz** [comment]
@vchudnov-g are we good to merge/release this yet? Thanks

**Comment by vchudnov-g** [comment]
@noahdietz No, let's not merge this yet. The server changes have not been rolled out yet.

File: README.md [hunk 2]
```diff
@@ -148,6 +156,15 @@ following attributes:
 
   * `metadata`: if `True`, [GapicMetadata](https://github.com/googleapis/googleapis/blob/master/gapic/metadata/gapic_metadata.proto) will be generated in JSON form. The default is `False`.
 
+  * `transport`: the desired transport(s) to generate, delimited by `+` e.g. `grpc+rest`.
+    * Acceptable values are `grpc` and `rest`.
+    * Defaults to `grpc`.
+
+  * `rest_numeric_enums`: if `True`, enables generation of system parameter requesting
```
Comment (vchudnov-g) [inline README.md:163]: Is this implemented in the Bazel rule already?

Comment (noahdietz) [reply README.md:163]: Yes, in this PR

**Review by vchudnov-g — APPROVED** [review]

**Pull request merged**